src/cursor/common.rs

```diff
@@ -200,3 +223,23 @@ pub(crate) struct CursorInformation {
     pub(crate) batch_size: Option<u32>,
     pub(crate) max_time: Option<Duration>,
 }
+
+/// Reads a `getMore` response form the server without sending a `getMore` command.
+pub(super) async fn read_exhaust_get_more(
```

Comment:
I worry that implementing this in such a direct manner outside of the executor could miss out on some required functionality. For instance, should we be updating the client and sessions' cluster times based off of get more responses? Also, in the event of network failure, we need to update the topology and clear the connection pool, which I don't think we can do here. Lastly, there seem to be tests verifying this behavior, but shouldn't we be emitting command monitoring events for these (perhaps with mocked command started events)? We probably need command failed events at least to ensure errors are properly logged.

Reply:
All of this fair. The difficulty is that in the absence of the specification actually being written, we don't know what the answers to any of these questions are. The cluster times might not make sense to keep track of, as the time that we're reading the responses could be much later than when the server sent them. Ditto with network failure; if we can't reach the server, that might be because the server _did_ shut down, and that's why it sent the responses in exhaust mode rather than waiting for getMore commands. Command monitoring is specifically an open question that the scope says the design will deal with, but the design is not being prioritized, so we don't know what the solution will be.

Despite all this, I still think this PR is worth merging because large parts of it will remain the same whenever the design does get written, and it will be less work to keep the code up to date by having it compiled as part of CI rather than trying to keep an external branch up to date through rebasing or letting it get stale and doing one giant rebase later. Maybe the best solution is to pre-emptively file a ticket for implementing the design that will eventually be written and then add a TODO comment here explaining that all these issues will need to be addressed once the spec work is done.

Reply:
That sounds good to me.